frontend/common/src/components/tabs/TabSet.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+import { TabProps } from "./Tab";
+import { TabColor } from ".";
+
+export interface TabSetProps extends React.HTMLProps<HTMLElement> {
+  /** The style type of the element. */
+  color?: TabColor;
+  children: React.ReactElement<TabProps>[];
+}
+
+interface TabSetState {
+  selectedTab: number;
+  isOpen: boolean;
+}
+
+const firstSelectableTab = (tabs: React.ReactElement<TabProps>[]): number => {
+  const firstIndex = tabs.findIndex(
+    (tab: React.ReactElement<TabProps>) =>
+      tab.props.variant == null || tab.props.variant === "normal",
+  );
+  return firstIndex >= 0 ? firstIndex : 0;
+};
+
+export const TabSet: React.FC<TabSetProps> = ({
+  color = "primary",
+  children,
+}): React.ReactElement => {
+  const [tabSetState, setTabSetState] = useState<TabSetState>({
+    selectedTab: firstSelectableTab(children),
+    isOpen: true,
+  });
+
+  const handleTabSelect = (index: number) => {
+    setTabSetState({
+      ...tabSetState,
+      isOpen: true,
+      selectedTab: index,
+    });
+  };
+  const handleToggleOpen = () => {
+    setTabSetState({
+      ...tabSetState,
+      isOpen: !tabSetState.isOpen,
+    });
+  };
+
+  // https://stackoverflow.com/a/32371612
+  const childrenWithProps = React.Children.map(children, (child, index) => {
+    // Checking isValidElement is the safe way and avoids a typescript
+    // error too.
+    if (React.isValidElement(child)) {
+      return React.cloneElement(child, {
+        isTabSetOpen: tabSetState.isOpen,
+        isTabSelected: tabSetState.selectedTab === index,
+        onSelect: () => handleTabSelect(index),
+        onToggleOpen: () => handleToggleOpen(),
+        color,
+      });
+    }
+    return child;
+  });
+
+  return (
+    <>
+      <div
+        data-h2-display="b(flex)"
+        data-h2-align-items="b(center)"
+        data-h2-margin="b(bottom, s)"
+      >
+        {childrenWithProps}
+      </div>
+      {tabSetState.isOpen && (
```

Comment:
Do we remove the closed tab from the DOM like this or do we need to do a visibility style instead?

Reply:
Switched to display:none as per https://github.com/GCTC-NTGC/gc-digital-talent/issues/2037#issuecomment-1040577534